services/retention/service.go

```diff
@@ -37,12 +49,23 @@ func NewService(c Config) *Service {
 	}
 }
 
+// OSSDropShardMetaRef creates a closure appropriate for OSS to use as DropShardMetaRef.
```

Comment:
Do we need this function, or could we simply use `mc.DropShard` as `DropShardMetaRef`?

Reply:
See example here: https://play.golang.com/p/IQo6LwLkpIR

Reply:
With the extra argument, this simplification becomes impossible, so disregard comment above.